components/server/src/prometheus-metrics.ts

```diff
@@ -151,3 +151,25 @@ const instanceStartsFailedTotal = new prometheusClient.Counter({
 export function increaseFailedInstanceStartCounter(reason: "clusterSelectionFailed" | "startOnClusterFailed") {
     instanceStartsFailedTotal.inc({ reason });
 }
+
+const prebuildsStartedTotal = new prometheusClient.Counter({
+    name: "gitpod_prebuilds_started_total",
+    help: "Counter of total prebuilds started.",
+    labelNames: ["state"],
+    registers: [prometheusClient.register],
+});
+
+export function increasePrebuildsStartedCounter(state: string) {
+    prebuildsStartedTotal.inc({ state });
+}
+
+const prebuildsEndedTotal = new prometheusClient.Counter({
+    name: "gitpod_prebuilds_ended_total",
+    help: "Counter of total prebuilds ended.",
+    labelNames: ["state"],
```

Comment:
We run multiple replicas of ws-manager-bridge, this means you'll get this metric reported from multiple pods. But not all are `governing` - that means actually affecting the database. This presents a challenge as your data here is not going to reflect reality, and will be `n` times larger depending on the number of replicas. One way to address that is to use the `governing` flag `ws-manager-bridge` has. You can for example add it as a label which will allow you to filter out the extra instances, or better, only increment the metric on governing instances.

Reply:
The `writeToDB` flag being true is what makes an instance governing.

Reply:
> The `writeToDB` flag being true is what makes an instance governing.

TIL!

Reply:
> We run multiple replicas of ws-manager-bridge

... but only ever one "bridge" is "governing" a given cluster. Thus we don't care about that flag, but should write the metric based on it. See [comment](https://github.com/gitpod-io/gitpod/pull/10435#discussion_r891991889).

Reply:
Basically what we're saying is that we should have
```
if (writeToDB) {
	reportMetric()
}
```